RelationalAI/Services/Client.cs

```diff
@@ -111,21 +111,27 @@ public async Task<DeleteDatabaseResponse> DeleteDatabaseAsync(string database)
             return Json<DeleteDatabaseResponse>.Deserialize(resp);
         }
 
-        public async Task<Engine> CreateEngineAsync(string engine, EngineSize size = EngineSize.XS)
+        public async Task<Engine> CreateEngineAsync(
+            string engine,
+            EngineSize size = EngineSize.XS,
+            Dictionary<string, string> customHeaders = null)
```

Comment:
Since we are tackling this now let make it generic, something like
```
public async Task<Engine> CreateEngineAsync(
            string engine,
            EngineSize size = EngineSize.XS,
            Dictionary<string, object> params = null)
            {
            if (params.ContainsKey("extraHeaders") ...
            or maybe params.TryGetValue ...
```
@bradlo what do you think ?

Reply:
The Rest object functions already take headers Dictionary<string, string>. So, the custom headers are being passed as headers to the Rest function directly. I don't know what will be the use of other params?

Reply:
sorry for the delay, had an issue with my ISP :pray: 
We want users to be able to overwrite all HTTP client parameters including headers (probably other parameters), the issue is that it's still not clear how to do so.
For julia and python we use kwargs which is different from the current implementation

Reply:
@NRHelmi  How about adding an overloaded function to create engine with a specific version?

Reply:
@miazamrai works for me :pray: